File: frontend/src/stores/product.ts
import { defineStore } from 'pinia'
import { ref } from 'vue'

export interface Product {
  id: number
  name: string
  price: number
  image: string
  category: string
  description: string
  stock: number
  rating: number
  sales: number
}

export const useProductStore = defineStore('product', () => {
  const products = ref<Product[]>([])
  const keyword = ref('')
  const loading = ref(false)

  return { products, keyword, loading }
})

File: backend/src/main/java/com/techstore/model/Product.java
package com.techstore.model;

import jakarta.persistence.*;
import lombok.Data;

@Data
@Entity
@Table(name = "products")
public class Product {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String name;
    private Double price;
    private String image;
    private String category;
    @Column(columnDefinition = "TEXT")
    private String description;
    private Integer stock;
    private Double rating;
    private Integer sales;
}
